include/directx/d3dx12.h

```diff
@@ -4114,18 +4152,35 @@ class CD3DX12FeatureSupport
         // Initialize per-node feature support data structures
         const UINT uNodeCount = m_pDevice->GetNodeCount();
         m_dProtectedResourceSessionSupport.resize(uNodeCount);
+        m_dArchitecture.resize(uNodeCount);
         m_dArchitecture1.resize(uNodeCount);
         for (UINT i = 0; i < uNodeCount; i++) {
-            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_PROTECTED_RESOURCE_SESSION_SUPPORT, m_dProtectedResourceSessionSupport[i]);
-            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i]);
+            m_dProtectedResourceSessionSupport[i].NodeIndex = i;
+            if (INITIALIZE_FAILED(D3D12_FEATURE_PROTECTED_RESOURCE_SESSION_SUPPORT, m_dProtectedResourceSessionSupport[i])) {
+                m_dProtectedResourceSessionSupport[i].Support = D3D12_PROTECTED_RESOURCE_SESSION_SUPPORT_FLAG_NONE;
+            }
+
+            m_dArchitecture[i].NodeIndex = i;
+            if (INITIALIZE_FAILED(D3D12_FEATURE_ARCHITECTURE, m_dArchitecture[i])) {
+                m_dArchitecture[i].TileBasedRenderer = false;
+                m_dArchitecture[i].UMA = false;
+                m_dArchitecture[i].CacheCoherentUMA = false;
+            }
+
+            m_dArchitecture1[i].NodeIndex = i;
+            if (INITIALIZE_FAILED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i])) {

```

Comment:
I think the pattern I'd use here would be:

```c++
if (INITIALIZE_FAILED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i])) {
    D3D12_FEATURE_DATA_ARCHITECTURE localArch0 = {};
    localArch0.NodeIndex = i;
    if (INITIALIZE_FAILED(D3D12_FEATURE_ARCHITECTURE, localArch0)) {
        localArch0.TileBasedRenderer = false;
        localArch0.UMA = false;
        localArch0.CacheCoherentUMA = false;
    }

    m_dArchitecture1[i].TileBasedRenderer = localArch0.TileBasedRenderer;
    m_dArchitecture1[i].UMA = localArch0.UMA;
    m_dArchitecture1[i].CacheCoherentUMA = localArch0.CacheCoherentUMA;
    m_dArchitecture1[i].IsolatedMMU = false;
}
```

So that way, we only have one copy of the architecture data in the structure, and we don't need to query D3D for both architecture structs. If the 1 version is supported, it's a superset of the older version and that's all we need to do. If the 1 version's not supported, we can query the older version, and copy the relevant fields to the 1 version, and fill in defaults for the stuff that's only in the newer version.

Reply:
Great idea! Updated.